tree/dataframe/inc/ROOT/RDF/RDisplay.hxx

```diff
@@ -90,17 +91,23 @@ private:
    size_t fNMaxCollectionElements = 10; // threshold on number of elements in collections to be Print()
 
    ////////////////////////////////////////////////////////////////////////////
-   /// Appends a cling::printValue call to the stringstream.
+   /// Appends a cling::printValue call to the stringstream or a string if the T type
+   ///    is an arithmetic type.
    /// \tparam T the type of the event to convert
    /// \param[in] stream Where the conversion function call will be chained.
    /// \param[in] element The event to convert to its string representation
    /// \param[in] index To which column the event belongs to
    /// \return false, the event is not a collection
-   template <typename T, std::enable_if_t<!ROOT::Internal::RDF::IsDataContainer<T>::value, int> = 0>
+   template <typename T, typename std::enable_if_t<!ROOT::Internal::RDF::IsDataContainer<T>::value, int> = 0>
    bool AddInterpreterString(std::stringstream &stream, T &element, const int &index)
    {
-      stream << "*((std::string*)" << ROOT::Internal::RDF::PrettyPrintAddr(&(fRepresentations[index]))
-             << ") = cling::printValue((" << fTypes[index] << "*)" << ROOT::Internal::RDF::PrettyPrintAddr(&element) << ");";
+      if (!std::is_arithmetic_v<T>) {
+         stream << "*((std::string*)" << ROOT::Internal::RDF::PrettyPrintAddr(&(fRepresentations[index]))
+                << ") = cling::printValue((" << fTypes[index] << "*)" << ROOT::Internal::RDF::PrettyPrintAddr(&element) << ");";
+      } else {
+         stream << "*((std::string*)" << ROOT::Internal::RDF::PrettyPrintAddr(&(fRepresentations[index]))
+                << ") = std::to_string(*(" << fTypes[index] << "*)" << ROOT::Internal::RDF::PrettyPrintAddr(&element) << ");";
+      }
```

Comment:
@vepadulano - the tests fail because `cling::printValue` and `std::to_string ` use different number of significant figures for floating point numbers:
`2.000000` vs `2.0000000`